Assets/Scripts/EnemyBehaviour.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EnemyBehaviour : MonoBehaviour
+{
+    [SerializeField] private float moveSpeed;
+    [SerializeField] private float jumpForce;
+    
+    private float _xVector;
+    private Animator _animator;
+    private Rigidbody2D _rigidbody2D;
+    private void Start()
+    {
+        _animator = GetComponent<Animator>();
+        _rigidbody2D = GetComponent<Rigidbody2D>();
+        _xVector = -1f;
+    }
+
+    private void Update()
+    {
+        Walk();
+    }
+
+    private void Walk()
+    {
+        float horizontalMovement = _xVector * moveSpeed * Time.deltaTime;
+        transform.Translate(new Vector3(horizontalMovement, 0, 0));
+
+        var hit = Physics2D.Raycast(transform.position, _xVector * transform.right, 0.5f,
+            LayerMask.GetMask("Ground"));
+        if (hit.collider != null)
+        {
+            Flip();
+        }
+    }
+
+
+    private void Flip()
+    {
+        _xVector = -_xVector; // Change the direction by negating _xVector
```

Comment:
you think that the people who are reading the code are dumb? no need for comments if your code is clear